### DISCUSSION TITLE
unable to run ImageRepository on azure... failed to configure authentication option...

### DISCUSSION THREAD

**Discussion by girishgouda** [fluxcd/flux2#3698] [General]
This is the yamlfile which i am using for ImageRepository in fluxv2 

apiVersion: image.toolkit.fluxcd.io/v1beta1
kind: ImageRepository
metadata:
  name: frontendapprepo
  namespace: flux-system
spec:
  secretRef:
    name: regcred
  image: docker.io/girishgoudar/voteapp
  interval: 1m0s

and this being the image policy 

 apiVersion: image.toolkit.fluxcd.io/v1beta1
kind: ImagePolicy
metadata:
  name: frontendapppolicy
  namespace: flux-system
spec:
  imageRepositoryRef:
    name: frontendapprepo
  policy:
    semver:
      range: 1.0.x                

![image](https://user-images.githubusercontent.com/9015874/226115210-0f783b2a-dbf0-4b25-863f-e5090ad4aa1a.png)

i have verified that i was able to pull the docker image from the secret created

**Comment by makkes** [comment]
Your Secret type is wrong, just as the error message says. It must be `kubernetes.io/dockerconfig`, not `kubernetes.io/dockercfg`.

**Reply by girishgouda** [reply]
i tried but i see the error for kubernetes.io/dockerconfig

![image](https://user-images.githubusercontent.com/9015874/226185440-fd48d441-3faa-4a2f-81ff-b581f5575dea.png)

can you let me know how to create the secret

**Reply by girishgouda** [reply]
i used the following yaml 

apiVersion: v1
kind: Secret
metadata:
  name: regcred
  namespace: flux-system
type: kubernetes.io/dockerconfig
data:
  .dockerconfigjson: base64 string

**Reply by makkes** [reply]
I'm sorry, type needs to be `kubernetes.io/dockerconfigjson`. My bad. 🤦🏻‍♂️ 

Documented [here](https://fluxcd.io/flux/components/image/imagerepositories/#secret-reference).

**Reply by girishgouda** [reply]
still facing issue with the auth 

![image](https://user-images.githubusercontent.com/9015874/226425860-8cf1c5f1-eb84-4d07-9658-10a317a4fc12.png)

can you provide a sample config.json for dockerhub?

**Comment by kingdonb** [comment]
Have you found this reference about ambient credentials:
https://fluxcd.io/flux/guides/image-update/#imagerepository-cloud-providers-authentication

It's generally expected you'll configure some ambient credential, also this is a relatively new feature so please be sure you're on the latest version of Flux; the `spec.provider` is a new feature, not in `v1beta1` API

(This field will not be available in the resource that you have without upgrading Flux and upgrading to the new API version in the resource.)

I just noticed that your image is in Docker Hub, so maybe your issue has nothing to do with Azure ambient credentials?

```
kubectl create secret docker-registry docker-pull-secret \
    --docker-username=<USER> \
    --docker-password=<PASSWORD> \
    --docker-server="index.docker.io"
```

Try creating your pull secret like this, if you're just trying to use Docker Hub then your issue likely has nothing to do with being on Azure.
